railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt

```diff
@@ -9,9 +9,19 @@ Rails.application.configure do
   # Do not eager load code on boot.
   config.eager_load = false
 
-  # Show full error reports and disable caching.
+  # Show full error reports.
   config.consider_all_requests_local       = true
-  config.action_controller.perform_caching = false
+
+  # Enable/disable caching. By default caching is disabled.
+  if Rails.root.join('tmp/caching-dev.txt').exist?
+    config.action_controller.perform_caching = true
+    config.static_cache_control = "public, max-age=172800"
+    config.cache_store = :memory_store
+  else
+    config.action_controller.perform_caching = false
+    config.cache_store = :null_store
+  end
```

Comment:
I'm not sure we should have such complex logic in the development environment file. But I don't really have better ideas.